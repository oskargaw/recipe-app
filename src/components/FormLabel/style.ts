import styled from "styled-components";

export const StyledFormLabel = styled.label`
  font-size: 1.5rem;
  font-weight: 500;
`;
