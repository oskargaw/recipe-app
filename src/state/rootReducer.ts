import { combineReducers } from "redux";
import { recipesReducer } from "./recipes/recipes.reducer";

const rootReducer = combineReducers({
  recipes: recipesReducer,
});

export type RootState = ReturnType<typeof rootReducer>;

export default rootReducer;
