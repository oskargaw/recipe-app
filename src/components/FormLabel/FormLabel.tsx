import { StyledFormLabel } from "./style";

export interface FormLabelProps {
  htmlFor: string;
  children: React.ReactNode;
}

export const FormLabel: React.FC<FormLabelProps> = ({ htmlFor, children }) => (
  <StyledFormLabel htmlFor={htmlFor}>{children}</StyledFormLabel>
);
