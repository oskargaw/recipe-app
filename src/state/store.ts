import { createStore } from "redux";
import { persistReducer, persistStore } from "redux-persist"; // imports from redux-persist
import storage from "redux-persist/lib/storage";
import rootReducer from "./rootReducer";

const persistConfig = {
  key: "root",
  storage,
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = createStore(
  persistedReducer,
  (window as any).__REDUX_DEVTOOLS_EXTENSION__ &&
    (window as any).__REDUX_DEVTOOLS_EXTENSION__()
);

const persistor = persistStore(store as any);

export { store, persistor };
