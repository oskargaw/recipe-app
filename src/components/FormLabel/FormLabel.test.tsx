import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { RootState } from "../../state/rootReducer";
import { FormLabel, FormLabelProps } from "./FormLabel";

describe("<FormLabel />", () => {
  const mockStore = configureStore([]);
  const initialState: RootState = {
    recipes: {
      recipeList: [],
    },
  };
  const store = mockStore(initialState);

  const props: FormLabelProps = {
    htmlFor: "test",
    children: "test",
  };

  it("should render correctly", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <FormLabel {...props} />
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
