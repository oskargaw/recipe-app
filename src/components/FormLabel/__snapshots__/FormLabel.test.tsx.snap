// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormLabel /> should render correctly 1`] = `
<DocumentFragment>
  <label
    class="sc-bdvvtL gJVqff"
    for="test"
  >
    test
  </label>
</DocumentFragment>
`;
